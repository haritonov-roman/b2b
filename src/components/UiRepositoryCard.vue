<script>
export default {
  name: 'UiRepositoryCard',

  props: {
    value: {
      type: Object,
      required: true,
      default: () => {}
    }
  }
}
</script>

<template>
  <cds-container
    color="primary"
    type="clickable"
    label="container"
    :href="value.route"
  >
    <div class="repository__wrap">
      <span class="repository__title">{{ value.name }}</span>
      <cds-icon name="External"/>
    </div>
  </cds-container>
</template>

<style lang="scss" scoped>
.repository {
  &__wrap {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__title {
    font-weight: var(--cds-font-margin-heading-4);
  }
}
</style>
