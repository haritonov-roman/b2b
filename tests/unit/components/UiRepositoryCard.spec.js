import { createLocalVue, shallowMount } from '@vue/test-utils'
import Cds from '@central-design-system/components'
import UiRepositoryCard from '@/components/UiRepositoryCard'

const localVue = createLocalVue()

localVue.use(Cds)

describe('UiRepositoryCard.vue', () => {
  const value = {
    name: 'Vue',
    route: 'vuejs/vue/issues'
  }

  const wrapper = shallowMount(UiRepositoryCard, {
    propsData: { value },
    localVue
  })

  it('Проверка рендеринга', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Проверка атрибута href', () => {
    expect(wrapper.attributes('href')).toMatch('vuejs/vue/issues')
  })

  it('Проверка заголовка карточки', () => {
    expect(wrapper.find('.repository__title').text()).toMatch('Vue')
  })

  it('Снимок', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
