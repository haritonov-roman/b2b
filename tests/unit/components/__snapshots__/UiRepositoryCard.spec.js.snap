// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiRepositoryCard.vue Снимок 1`] = `
<cds-container-stub
  color="primary"
  href="vuejs/vue/issues"
  label="container"
  tag="div"
  type="clickable"
>
  <div
    class="repository__wrap"
  >
    <span
      class="repository__title"
    >
      Vue
    </span>
     
    <cds-icon-stub
      name="External"
    />
  </div>
</cds-container-stub>
`;
